File: distros/J/JavonetPerlSdk/lib/Javonet/Binaries/Ruby/MacOs/X64/core/handler/enable_type_handler.rb
require_relative 'abstract_command_handler'
require_relative '../type_cache/type_cache'

class EnableTypeHandler < AbstractCommandHandler
  def initialize
    @required_parameters_count = 1
  end

  def process(command)
    begin
      if command.payload.length < @required_parameters_count
        raise "#{self.class.name} parameters mismatch!"
      end

      type_cache = TypeCache.instance

      command.payload.each do |payload|
        if payload.is_a? String
          type_cache.cache_type(payload)
        elsif payload.is_a? Array
          payload.each do |type_to_enable|
            type_cache.cache_type(type_to_enable)
          end
        end

      end
      0
    end
  rescue Exception => e
    return e
  end
end

File: distros/J/JavonetPerlSdk/lib/Javonet/Binaries/Python/javonet/core/handler/EnableTypeHandler.py
from javonet.core.handler.AbstractCommandHandler import AbstractCommandHandler
from javonet.core.typeCache.TypeCache import TypeCache


class EnableTypeHandler(AbstractCommandHandler):
    def __init__(self):
        self._required_parameters_count = 1

    def process(self, command):
        try:
            if len(command.payload) < self._required_parameters_count:
                raise Exception(self.__class__.__name__ + " parameters mismatch!")

            type_cache = TypeCache()

            for payload in command.payload:
                if isinstance(payload, str):
                    type_cache.cache_type(payload)
                if isinstance(payload, list):
                    for type_to_enable in payload:
                        type_cache.cache_type(type_to_enable)

            return 0

        except Exception as e:
            exc_type, exc_value = type(e), e
            new_exc = exc_type(exc_value).with_traceback(e.__traceback__)
            raise new_exc from None
